src/components/Notification/Notification.tsx

```diff
@@ -98,14 +107,45 @@ export const Notification = ({
 
   const [width, setWidth] = useState("");
 
-  useEffect(() => {
+  const [isTruncated, setIsTruncated] = useState(true);
+  const [needsTruncation, setNeedsTruncation] = useState(false);
+
+  const notificationTranslations = useMemo(() => {
+    return {
+      ...defaultTranslations,
+      ...translations,
+    };
+  }, [translations]);
+
+  function handleResize() {
```

Comment:
nit: for consistency I recommend using arrow methods
```suggestion
  const handleResize = () => {
```

Reply:
Actually ended up moving this method out of the Component as I had to refactor the logic to resolve some issues when using the PopUpManager for non-inline Notifications

As an aside, I think that makes this and other methods in the file easier to unit tests, although I haven't added any tests for them in this PR as it is already quite large. I think I will either create a third PR for this, or come back and add these tests after completing the work in the CSS library, as this ticket has already ballooned quite a bit due to changing AC